Tests/TuistKitTests/Generator/ManifestTargetGeneratorTests.swift

```diff
@@ -0,0 +1,50 @@
+import Basic
+import Foundation
+import TuistCore
+import XCTest
+@testable import TuistKit
+
+class ManifestTargetGeneratorTests: XCTestCase {
+    func test_generateManifestTarget() throws {
+        // Given
+        let path = AbsolutePath("/test")
+        let manifestLoader = MockGraphManifestLoader()
+        manifestLoader.manifestPathStub = { _, _ in
+            path.appending(component: "Project.swift")
+        }
+        let resourceLocator = MockResourceLocator()
+        resourceLocator.projectDescriptionStub = {
+            AbsolutePath("/test/ProjectDescription.dylib")
+        }
+
+        let subject = ManifestTargetGenerator(manifestLoader: manifestLoader,
+                                              resourceLocator: resourceLocator)
+        // When
+        let target = try subject.generateManifestTarget(for: "MyProject", at: path)
+
+        // Then
+        XCTAssertEqual(target.name, "MyProject-Manifest")
+        XCTAssertEqual(target.sources.map { $0.asString }, ["/test/Project.swift"])
+        XCTAssertNil(target.infoPlist)
+        assertValidManifestBuildSettings(for: target)
+    }
+
+    // MARK: - Helpers
+
+    func assertValidManifestBuildSettings(for target: Target) {
```

Comment:
I'd make it `private`, and maybe mark the class (and mocks) as `final` (to help the compiler move faster - at least in theory).
Maybe it's worth to change the locations a bit, currently everything is in `/test` and does not reflect the real world cases and it's harder to understand where does the `XCTAssertEqual(settings.base["FRAMEWORK_SEARCH_PATHS"], "/test")` path come from. Not sure, what about "/products/ProjectDescription.dylib" and maybe the path can be passed to the assertion as well?

Reply:
> I'd make it private, and maybe mark the class (and mocks) as final (to help the compiler move faster - at least in theory).

Agree here.

> Maybe it's worth to change the locations a bit, currently everything is in /test and does not reflect the real world cases and it's harder to understand where does the XCTAssertEqual(settings.base["FRAMEWORK_SEARCH_PATHS"], "/test") path come from.

I think it's ok in this case to keep the path `/test`. Although we know `/test` won't be the real path, we know for sure that `resourceLocator` will gives us that information, and that's why we have a mock for it.